core/src/main/java/io/undertow/server/handlers/SameSiteCookieHandler.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.undertow.server.handlers;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+import io.undertow.UndertowMessages;
+import io.undertow.server.HandlerWrapper;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.ResponseCommitListener;
+import io.undertow.server.handlers.builder.HandlerBuilder;
+import io.undertow.util.Headers;
+import io.undertow.util.SameSiteNoneIncompatibleClientChecker;
+
+/**
+ * Handler that will set the SameSite flag on all cookies
+ */
+public class SameSiteCookieHandler implements HttpHandler {
+
+    private static final Logger log = Logger.getLogger(SameSiteCookieHandler.class);
+
+    private final HttpHandler next;
+    private final String mode;
+    private final String cookieName;
+    private final boolean disableClientChecker;
+
+    public SameSiteCookieHandler(final HttpHandler next, final String mode) {
+        this(next, mode, null, false);
```

Comment:
do we want to have a disable client checker with a default value set to false, or do we want to have a enable-client-checker with a default value set to true? I wonder how does this work in the long term pov, do we expect to still have a long list of incompatible user agents in the future?

Reply:
I have no preference whether disable-client-checker (false by default) or enable-client-checker (true by default). 

I'm not sure how long we should keep the client checker logic. Future user-agents will not be incompatible, but I guess some old incompatible user-agents (e.g. mobile browsers on Android or iOS) can remain for a longer time than we expect. So, I think we should enable the client checker by default.

Reply:
I wonder if this should be enabled by default. If the browser is old and does not support "None", it will fall back to ignoring the attribute completely in case it finds a SameSite=None. In the old specs, it says that anything different from Lax or Strick should cause the attribute to be ignored. Isn't this the same behavior we would get from a browser that supports SameSite="None" anyway?

Reply:
Please, ignore. I see that some browser versions were rejecting the cookie instead of ignoring it.